pkg/vm/context.go

```diff
@@ -45,6 +45,14 @@ type Context struct {
 	// Call flags this context was created with.
 	callFlag smartcontract.CallFlag
 
+	// InvocationState contains expected return type and actions to be performed on context unload.
+	InvocationState
+}
+
+// InvocationState contains return convention and callback to be executed on context unload.
+type InvocationState struct {
+	// Callback is executed on context unload.
+	Callback func(ctx *Context)
```

Comment:
Do we really need it?

Reply:
We don't need it now, though it exists in C# code.
If we remove it, some tests for native contract will also be removed. I am ok with both variants, as we have it covered with tests.

Reply:
I think we better not introduce things not need for the protocol to work. I'd say it's an internal C# thingie (that's not really used even there), so we can not complicate our code with it.